lib/post_creator.rb

```diff
@@ -366,6 +366,9 @@ def update_topic_auto_close
 
   def setup_post
     @opts[:raw] = TextCleaner.normalize_whitespaces(@opts[:raw] || '').gsub(/\s+\z/, "")
+    if SiteSetting.topic_featured_link_enabled && SiteSetting.topic_featured_link_style == 'onebox'
+      @opts[:raw] = DiscourseFeaturedLink.cache_onebox_link(@opts[:featured_link])
```

Comment:
In a category that doesn't allow featured link topics, and the post raw is one line with a link to a jpg (for example), this line will set raw to nil and exception will be thrown.